rules/library.bzl

```diff
@@ -439,6 +451,24 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
     namespace = module_name if namespace_is_module_name else name
     module_map = kwargs.pop("module_map", None)
     swift_objc_bridging_header = kwargs.pop("swift_objc_bridging_header", None)
+
+    # Historically, xcode and cocoapods use an umbrella header that imports Foundation and UIKit at the
+    # beginning of it. See:
+    # * https://github.com/CocoaPods/CocoaPods/issues/6815#issuecomment-330046236
+    # * https://github.com/facebookarchive/xcbuild/issues/92#issuecomment-234372926
+    #
+    # As a result, when writing swift code, there is no need for importing neither Foundation nor
+    # UIKit. See:
+    # * https://github.com/facebookarchive/xcbuild/issues/92#issuecomment-234400427
+    #
+    # But these automatic imports are a problem when strict imports are wanted. See:
+    # * https://forums.swift.org/t/supporting-strict-imports/42472
+    #
+    # So provide here two behaviours:
+    # * By default, follow xcode and cocoapods and populate the umbrella header with the usual content
```

Comment:
@acecilia thank you for adding this! This is definitely less than ideal for many reasons as a default IMO. Longer term we can probably set it to false. The problem is that much iOS code was written to omit those imports - so that code still needs it. 

My main suggestion is to rename it from `legacy` to `generate_sdk_umbrella_header` or `generate_default_umbrella_header` , unless the "legacy" nomenclature is aligned with a cocoapods name

Reply:
Done ✌️  Renamed to `generate_default_umbrella_header`

Reply:
BTW, some extra context. Yes I had to go around the full codebase adding `import Foundation` and `import UIKit` where needed 💩 . But I did not do it manually. I wrote a script that:

1. Builds the codebase with bazel
2. Captures the output, filtering the lines saying ` error: cannot find`
3. Collect the files with above error
4. Adds `import Foundation` on the top of the file, or `import UIKit` if `import Foundation` is found already
5. Back to step 1